android/app/src/main/java/com/coboltforge/dontmind/multivnc/EditBookmarkActivity.java

```diff
@@ -66,7 +65,8 @@ public void onCreate(Bundle savedInstanceState) {
 		// read connection from DB
 		Intent intent = getIntent();
 		long connID = intent.getLongExtra(Constants.CONNECTION, 0);
-		if (bookmark.Gen_read(database.getReadableDatabase(), connID))
+		bookmark = database.getConnectionDao().get(connID);
```

Comment:
Does the DAO method return null when no entry is found? Could not find any docu on this...

Reply:
Yes. I had to check the generated implementation for DAO to be sure about this. Following links sort-of confirmed this:

https://developer.android.com/reference/androidx/room/Query
https://developer.android.com/reference/androidx/room/EmptyResultSetException